src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -266,23 +275,207 @@ void AtlasEngine::_drawGlyph(const AtlasQueueItem& item) const
+            /* mappedLength       */ &mappedLength,
+            /* mappedFont         */ mappedFont.addressof(),
+            /* scale              */ &mappedScale));
+
+        if (mappedFont)
+        {
+            wil::com_ptr<IDWriteFontFace> fontFace;
+            THROW_IF_FAILED(mappedFont->CreateFontFace(fontFace.addressof()));
+
+            DWRITE_FONT_METRICS metrics;
+            fontFace->GetMetrics(&metrics);
+
+            const u32 codePoint = key->chars[0];
+            u16 glyphIndex;
+            THROW_IF_FAILED(fontFace->GetGlyphIndicesW(&codePoint, 1, &glyphIndex));
+
+            DWRITE_GLYPH_METRICS glyphMetrics;
+            THROW_IF_FAILED(fontFace->GetDesignGlyphMetrics(&glyphIndex, 1, &glyphMetrics));
+
+            const f32x2 boxSize{
+                static_cast<f32>(glyphMetrics.advanceWidth) / static_cast<f32>(metrics.designUnitsPerEm) * _r.fontMetrics.fontSizeInDIP,
+                static_cast<f32>(glyphMetrics.advanceHeight) / static_cast<f32>(metrics.designUnitsPerEm) * _r.fontMetrics.fontSizeInDIP,
+            };
+
+            // We always want box drawing glyphs to exactly match the size of a terminal cell.
+            // So for safe measure we'll always scale them to the exact size.
+            // But add 1px to the destination size, so that we don't end up with fractional pixels.
+            scalingRequired = true;
+            scale.x = (layoutBox.x + _r.dipPerPixel) / boxSize.x;
+            scale.y = (layoutBox.y + _r.dipPerPixel) / boxSize.y;
+        }
+    }
+    else
+    {
+        DWRITE_OVERHANG_METRICS overhang;
+        THROW_IF_FAILED(textLayout->GetOverhangMetrics(&overhang));
+
+        const DWRITE_OVERHANG_METRICS clampedOverhang{
+            std::max(0.0f, overhang.left),
+            std::max(0.0f, overhang.top),
+            std::max(0.0f, overhang.right),
+            std::max(0.0f, overhang.bottom),
+        };
+        f32x2 targetSize{
+            layoutBox.x - _r.dipPerPixel,
+            layoutBox.y - _r.dipPerPixel,
+        };
+        f32x2 actualSize{
+            layoutBox.x + overhang.left + overhang.right,
+            layoutBox.y + overhang.top + overhang.bottom,
+        };
+
+        // Long glyphs should be drawn with their proper design size, even if that makes them a bit blurry,
+        // because otherwise we fail to support "pseudo" block characters like the "===" ligature in Cascadia Code.
+        // If we didn't force upscale that ligatures it would seemingly shrink shorter and shorter, as its
+        // glyph advance is often slightly shorter by a fractional pixel or two compared to our terminal's cells.
+        // It's a trade off that keeps most glyphs "crisp" while retaining support for things like "===".
+        // At least I can't think of any better heuristic for this at the moment...
+        if (cellCount > 2)
+        {
+            scalingRequired = true;
+            scale = _r.fontMetrics.scale;

```

Comment:
I just realized that this approach is slightly incorrect.
TODO (for myself): Set `scale` to the `max(.x, .y)` so that the glyph is resized while preserving the aspect ratio.